src/Service/EventLog/Loggers/EventCache.cs

```diff
@@ -0,0 +1,12 @@
+using Passwordless.Service.EventLog.Models;
+
+namespace Passwordless.Service.EventLog.Loggers;
+
+public class EventCache
```

Comment:
Is this cache used anywhere outside the EfWriter? If not - should it be a private subclass, not injected by DI? Seems to be to just be a implementation detail on how we do delayed writes to EF, no?

Reply:
By putting it in DI as scoped, it exists for the lifetime of the request. The request could pull different instances of the `IEventLogger` throughout the request depending on where we put logs.  This acts as a manager class I guess to hold the event collection for the lifetime of the request. The Unauthenticated Logger could be pulled, write an event for something, then move on. Later another thing could write an event and finish.  At the end of the request, we would ask for a final logger, and then commit everything from the Event Cache and clear it out. 

If we removed it from DI, then when the one instance was destroyed, then we'd lose the events held in that collection. Or am I missing something?